### PR TITLE
Fix decompress_peek_gz to cope with files starting on empty gzip blocks.

### PR DESCRIPTION
**Pull request by jkbonfield**
Demonstration:

```
$ echo -e "@seq\nAAAAA\n+\nABCDE" | gzip > _normal.gz                           
$ (gzip < /dev/null;cat _normal.gz) > _empty.gz                                 
                                                                                
$ ./htsfile _normal.gz                                                          
_normal.gz:     FASTQ gzip-compressed sequence data                             
                                                                                
$ ./htsfile _empty.gz                                                           
_empty.gz:      empty gzip-compressed data                                      
$ ./test/test_view _empty.gz                                                    
Unsupported or unknown category of data in input file                           
```

Zlib's `inflate` ends with `Z_STREAM_END` and then repeated calls to get more won't give any difference, so we need the explicit reset to move forwards with the next block.

### CODE REVIEW

**Comment by jkbonfield** [comment]
Now with added paranoia for loop detection.  Just incase...

File: hts.c [hunk 1]
```diff
@@ -324,10 +324,24 @@ decompress_peek_gz(hFILE *fp, unsigned char *dest, size_t destsize)
...
+                last_in = zs.next_in;
+            inflateReset(&zs);
+            continue;
+        }
+        if (ret != Z_OK)
+            break;
+    }
 
-    destsize = zs.total_out;
+    // zs.total_out can sometimes be wrong as inflateReset resets it
```
Comment (jmarshall) [inline hts.c:343]: If `total_out` may have been reset by `inflateReset`, is it still appropriate to use it in the loop above?

Comment (jkbonfield) [reply hts.c:343]: Doh!  Good point.  No (although in practice avail_in seems to catch it).

I'll make a follow up fix with the similar `next_out - dest` logic instead.

Comment (jkbonfield) [reply hts.c:343]: It appears I was being overly paranoid (although I left in the loop detection bit, even though I can't actually trigger zlib to consume no input without producing an error).

`avail_out` would have been the correct variable, as unlike `total_out` this isn't reset.  However it turns out neither are necessary.  Calling `inflate` with one of those as zero returns an error (Z_BUF_ERROR) so it promptly bails out the loop anyway, and even if it just returned Z_STREAM_END again it'd be caught by the other checks.

So it's now much simpler.  See #1646 for 2nd try at fixing this :)

**Pull request merged**